src/shortcuts/__tests__/extractThought.ts

```diff
@@ -0,0 +1,69 @@
+import { screen } from '@testing-library/react'
+import { extractThought, newThought } from '../../action-creators'
+import { store } from '../../store'
+import createTestApp, { cleanupTestApp } from '../../test-helpers/createTestApp'
+import { setCursorFirstMatchActionCreator } from '../../test-helpers/setCursorFirstMatch'
+
+/**
+ * Set range selection.
+ */
+const setSelection = (element: HTMLElement, selectionStart: number, selectionEnd: number) => {
+  const range = document.createRange()
+  const sel = window.getSelection()
+
+  range.setStart(element.childNodes[0], selectionStart)
+  range.setEnd(element.childNodes[0], selectionEnd)
+
+  sel?.removeAllRanges()
+  sel?.addRange(range)
+
+  return range.toString()
+}
+
+describe('Extract thought', () => {
+  beforeEach(async () => {
+    await createTestApp()
+  })
+  afterEach(cleanupTestApp)
+
+  it('the selected part of a thought is extracted as a child thought', async () => {
+
+    store.dispatch([
+      newThought({ value: 'this is a thought' }),
+      setCursorFirstMatchActionCreator(['this is a thought'])
+    ])
+
+    const thought = screen.queryAllByText('this is a thought')
+    expect(thought).toHaveLength(2)
```

Comment:
I wonder why this returns two instances. Is it the `Thought` and the `ThoughtAnnotation`? If so, should probably just select what is visible (`Thought`), though I am open to other ideas. Selecting a visible and a hidden instance and assuming the first one is the visible one feels like too much implementation detail leaking into the tests.

Reply:
I couldn't find a query that ignores hidden elements, so I've used a matching element selector.